pennylane_lightning/core/src/bindings/BindingsBase.hpp

```diff
@@ -76,4 +76,59 @@ void registerGatesForStateVector(PyClass &pyclass) {
         pyclass.def(gate_name.c_str(), func, doc.c_str());
     });
 }
+
+/**
+ * @brief Register matrix.
+ */
+template <class TensorNetT>
+void registerTensor(
+    TensorNetT &tensor_network,
+    const py::array_t<std::complex<typename TensorNetT::PrecisionT>,
+                      py::array::c_style | py::array::forcecast> &matrix,
+    const std::vector<std::size_t> &wires, bool inverse = false) {
+    using ComplexT = typename TensorNetT::ComplexT;
+    const auto m_buffer = matrix.request();
+    std::vector<ComplexT> conv_matrix;
+    if (m_buffer.size) {
+        const auto m_ptr = static_cast<const ComplexT *>(m_buffer.ptr);
+        conv_matrix = std::vector<ComplexT>{m_ptr, m_ptr + m_buffer.size};
+    }
+    tensor_network.applyOperation("applyMatrix", wires, inverse, {},
+                                  conv_matrix);
+}
+
+/**
+ * @brief Register TensorNet class to pybind.
+ *
+ * @tparam TensorNetT Tensor network type to register
+ * @tparam Pyclass Pybind11's class object type
+ *
+ * @param pyclass Pybind11's class object to bind tensor network
+ */
+template <class TensorNetT, class PyClass>
+void registerGatesForTensorNet(PyClass &pyclass) {
+    using PrecisionT = typename TensorNetT::PrecisionT; // TensorNet's precision
+    using ParamT = PrecisionT; // Parameter's data precision
+
+    using Pennylane::Gates::GateOperation;
+    using Pennylane::Util::for_each_enum;
+    namespace Constant = Pennylane::Gates::Constant;
+
+    pyclass.def("applyMatrix", &registerTensor<TensorNetT>,
+                "Apply a given matrix to wires.");
+
+    for_each_enum<GateOperation>([&pyclass](GateOperation gate_op) {
+        using Pennylane::Util::lookup;
+        const auto gate_name =
+            std::string(lookup(Constant::gate_names, gate_op));
+        const std::string doc = "Apply the " + gate_name + " gate.";
+        auto func = [gate_name = gate_name](
+                        TensorNetT &tensor_network,
+                        const std::vector<std::size_t> &wires, bool inverse,
+                        const std::vector<ParamT> &params) {
+            tensor_network.applyOperation(gate_name, wires, inverse, params);
+        };
+        pyclass.def(gate_name.c_str(), func, doc.c_str());
+    });
+}
```

Comment:
Is there a way to fuse that with the other registers we have for statevectors? We could maybe change the names to be more general, like `registerGates` for example. The header BindingsBase was designed to be common to all backends, or at least all statevector backends. If LightningTensor departs too much from our statevector backends, we can create a new base for its bindings, or move this to `BindingsLTensor.hpp`.

Reply:
Thanks folks! This is something definitely we can do but I would like to keep it as it is for now since we have to rename the method name here as @AmintorDusko mentioned above, which will leads to changes in the state vector backends. Add a TODO, let's refactor it later.